Deprecation notice of ambiguous `Proof using` and `Collection` usage (#15127)
coq/coq: doc/sphinx/proofs/writing-proofs/proof-mode.rst

         Collection Many := Fewer - (x y)

   .. deprecated:: 8.15

>       Redefining a collection, defining a collection with the same name as a variable,
>       and invoking the :cmd:`Proof using` command whenever names of collections and variables overlap
>       are deprecated. See the documentation of the warnings below and the warnigns
>       of the :cmd:`Proof using` command.

Reviewer: ```suggestion
      Redefining a collection, defining a collection with the same name as a variable,
      and invoking the :cmd:`Proof using` command when collection and variable names overlap
      are deprecated. See the warnings below and in the :cmd:`Proof using` command.
```
